### PR TITLE
[3.1] 1980681: Replace DELETE with TRUNCATE were possible [ENT-4065]

### PR DESCRIPTION
**Pull request by sonalidhome**

### CODE REVIEW

File: server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml [hunk 1]
```diff
@@ -7,21 +7,38 @@
...
-                WHERE EXISTS (SELECT cac.serial_id FROM cp_cont_access_cert cac WHERE cac.serial_id = cs.id);
+            WHERE EXISTS (SELECT cac.serial_id FROM cp_cont_access_cert cac WHERE cac.serial_id = cs.id);
         </sql>
 
         <update tableName="cp_consumer">
             <column name="cont_acc_cert_id"/> <!-- this sets the column value to null -->
         </update>
 
-        <delete tableName="cp_cont_access_cert" />
+        <!--Below changes are made to fix the bug #1980681. -->
```
Comment (Ceiu) [inline server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml:24]: Add a space after the comment tag open

File: server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml [hunk 2]
```diff
@@ -7,21 +7,38 @@
...
     <changeSet id="20201103112757-1" author="crog">
+        <validCheckSum>7:b060e85fd3270acbb9778b17c6d0a447</validCheckSum>
         <comment>
             Removes any existing SCA certs to address a problem with cert generation in some organizations
         </comment>
 
         <sql>
             UPDATE cp_cert_serial cs SET revoked = true
-                WHERE EXISTS (SELECT cac.serial_id FROM cp_cont_access_cert cac WHERE cac.serial_id = cs.id);
+            WHERE EXISTS (SELECT cac.serial_id FROM cp_cont_access_cert cac WHERE cac.serial_id = cs.id);
```
Comment (Ceiu) [inline server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml:17]: Keep that indentation, as it's a continuation of the previous statement.

File: server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml [hunk 3]
```diff
@@ -7,21 +7,38 @@
...
+                baseTableName="cp_consumer"
+                baseColumnNames="cont_acc_cert_id"
+                constraintName="fk_cont_acc_cert"
+                deferrable="false"
+                initiallyDeferred="false"
+                onDelete="NO ACTION"
+                onUpdate="NO ACTION"
+                referencedColumnNames="id"
+                referencedTableName="cp_cont_access_cert"
+                referencesUniqueColumn="false" />
```
Comment (Ceiu) [inline server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml:42]: Something I should have caught on the PR for master: these lines are indented an extra tab width.

**Review by Ceiu — CHANGES_REQUESTED** [review]
Minor formatting changes only. Manual testing checked out: was able to deploy directly and against a CP which already ran the original version of the changeset.

We'll want to forward port all of this to master as well, to cover the various oversights there.

File: server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml (hunk 1 above)
Comment (sonalidhome) [reply server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml]: Done.

File: server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml (hunk 2 above)
Comment (sonalidhome) [reply server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml]: Done.

File: server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml (hunk 3 above)
Comment (sonalidhome) [reply server/src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml]: Done.

**Review by Ceiu — APPROVED** [review]

**Pull request merged**